addrmgr/doc.go

```diff
@@ -1,39 +1,41 @@
+// Copyright (c) 2015-2024 The Decred developers
 // Use of this source code is governed by an ISC
 // license that can be found in the LICENSE file.
 
 /*
-Package addrmgr implements concurrency safe Decred address manager.
+Package addrmgr implements a concurrency-safe Decred address manager.
 
 # Address Manager Overview
 
-In order maintain the peer-to-peer Decred network, there needs to be a source
-of addresses to connect to as nodes come and go.  The Decred protocol provides
-the getaddr and addr messages to allow peers to communicate known addresses
-with each other.  However, there needs to a mechanism to store those results and
-select peers from them.  It is also important to note that remote peers can't
-be trusted to send valid peers nor attempt to provide you with only peers they
-control with malicious intent.
+The Decred network relies on fully-validating nodes that relay transactions and
+blocks to other nodes around the world. The network must be dynamic because
+nodes will connect and disconnect as they please. Each node must manage a source
+of IP addresses to connect to and share with other nodes. The Decred wire
+protocol provides the `getaddr` and `addr` messages, allowing peers to request
+and share known addresses with each other. Each node needs a way to store those
+addresses and select peers from them. However, it is important to remember that
+remote peers cannot be trusted. A remote peer might send invalid addresses, or
+worse, only send addresses they control with malicious intent.
 
-With that in mind, this package provides a concurrency safe address manager for
-caching and selecting peers in a non-deterministic manner.  The general idea is
-the caller adds addresses to the address manager and notifies it when addresses
-are connected, known good, and attempted.  The caller also requests addresses as
-it needs them.
+With that in mind, this package provides a concurrency-safe address manager for
+caching and selecting peers in a non-deterministic manner. The general idea is
+that the caller adds addresses to the address manager and notifies it when
+addresses are connected, known good, and attempted. The caller also requests
+addresses as it needs them.
 
 The address manager internally segregates the addresses into groups and
-non-deterministically selects groups in a cryptographically random manner.  This
-reduce the chances multiple addresses from the same nets are selected which
-generally helps provide greater peer diversity, and perhaps more importantly,
-drastically reduces the chances an attacker is able to coerce your peer into
-only connecting to nodes they control.
+non-deterministically selects groups in a cryptographically random manner. This
+reduces the chances of selecting multiple addresses from the same network, which
+generally helps provide greater peer diversity. More importantly, it drastically
+reduces the chances that an attacker could coerce your peer into connecting only
```

Comment:
Substituted with "can":

> More importantly, it drastically reduces the chances that an attacker can coerce your peer into connecting only to nodes they control.

Reply:
instead of the wishwashy "can", "could", why not actually describe the attack actually being pulled off?

> More importantly, it drastically reduces the chances that an attacker coerces your peer...

Reply:
I like getting rid of can/could. Moving in that direction, I'm updating to:

> More importantly, it drastically reduces the chances of an attacker coercing your peer into connecting only to nodes they control.

Reply:
I like that better too.  Good suggestion and update.